src/AddIns/Uno.WinUI.Graphics3DGL/GLCanvasElement.cs

```diff
@@ -140,6 +136,20 @@ protected GLCanvasElement(uint width, uint height, Func<Window>? getWindowFunc)
 		}
 	}
 
+	private void OnClosed(object o, WindowEventArgs windowEventArgs)
+	{
+		if (GlAvailable!.Value)
+		{
+			// OnUnloaded is called after OnClosed, which leads to disposing the context first and then trying to
+			// delete the framebuffer, etc. and this causes exceptions.
+			DispatcherQueue.TryEnqueue(DispatcherQueuePriority.Low, () =>
+			{
+				using var _ = _nativeOpenGlWrapper!.MakeCurrent();
+				_nativeOpenGlWrapper?.Dispose();
```

Comment:
if `_nativeOpenGlWrapper` was `null` here then the previous line will throw an NRE

Reply:
Nice catch. It should be guaranteed to not be null because of the outer check for GlAvailable. I removed the `?` and adjusted the comments.